### PR TITLE
Autofocus analyzer v1

### PR DESCRIPTION
**Pull request by amr-cossi**
#472 

- [x] Add docs to analyzer requirements in CortexDocs repo

### CODE REVIEW

**Comment by nadouani** [comment]
Hello @amr-cossi Thanks for the PR, we are reviewing it. 

Is it possible to add a PR for https://github.com/TheHive-Project/CortexDocs/blob/master/analyzer_requirements.md

**Comment by amr-cossi** [comment]
Yes I'll do it.

File: analyzers/Autofocus/requirements.txt [hunk 1]
```diff
@@ -0,0 +1,2 @@
+cortexutils
+git+https://github.com/PaloAltoNetworks/autofocus-client-library ; python_version<='2.7'
```
[the remark this answers is not in the document]
Comment (amr-cossi) [reply analyzers/Autofocus/requirements.txt]: The PR has been sent before this library was Python3 compatible. It is now and the analyzer runs in Python 3 on my instance. We can remove this python_version<='2.7' requirement and change the .py header to #!/usr/bin/env python3.

[the remark this answers is not in the document]
Comment (nadouani) [reply analyzers/Autofocus/requirements.txt]: Perfect.

**Comment by jeromeleonard** [comment]
Hello @amr-cossi, 

Do you have any idea about this issue pls (https://github.com/TheHive-Project/Cortex-Analyzers/issues/612#issuecomment-584762995)  ?

Made some tests with the lib on my side (without valid api key). I received the same error message `(ClientError(...), 'API key is not set. Library requires AutoFocusAPI.api_key to be set, or apikeyto be provided via configuration file.')`

```
>>> from autofocus import AFSample, AFConnectionActivity, AFUserAgentFragment, AFRelatedMacro, AutoFocusAPI
>>> AutoFocusAPI.api_key='blah'
>>> AFSample.get("8404e06ff383275462298e830bebe9540fab2092eca5523649d74e6e596ac23d")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/jerome/Devel/venv/Cortex-Analyzers3/lib/python3.7/site-packages/autofocus/models/sample.py", line 237, in get
    return SampleFactory().get(*args, **kwargs)
  File "/Users/jerome/Devel/venv/Cortex-Analyzers3/lib/python3.7/site-packages/autofocus/factories/sample.py", line 403, in get
    for sample in self.search(query, attributes=attributes):
  File "/Users/jerome/Devel/venv/Cortex-Analyzers3/lib/python3.7/site-packages/autofocus/factories/sample.py", line 190, in _not_coro
    for res in self._api_search("/samples/search", query, scope, sort_by, sort_order, fields, limit):
  File "/Users/jerome/Devel/venv/Cortex-Analyzers3/lib/python3.7/site-packages/autofocus/factories/base.py", line 735, in _not_coro
    for res in req_class(path, post_data=post_data, async_req=async_req).run():
  File "/Users/jerome/Devel/venv/Cortex-Analyzers3/lib/python3.7/site-packages/autofocus/factories/base.py", line 521, in __init__
    super().__init__(*args, **kwargs)
  File "/Users/jerome/Devel/venv/Cortex-Analyzers3/lib/python3.7/site-packages/autofocus/factories/base.py", line 217, in __init__
    raise ClientError("API key is not set. Library requires AutoFocusAPI.api_key to be set, or apikey"
autofocus.exceptions.ClientError: (ClientError(...), 'API key is not set. Library requires AutoFocusAPI.api_key to be set, or apikeyto be provided via configuration file.')
```


 Sounds like the value of the key is never read by the library.

**Comment by amr-cossi** [comment]
Hi,
Love breaking changes without documentation in libraries.
I confirm that it was working in v3.0.1 and it is broken since they have changed the way config in handled.
We can open an issue on the Github of the library but in the meantime, I have a quick workaround : 

`import os
os.environ["AUTOFOCUS_APIKEY"] = "blah"
from autofocus import AFSample, AFConnectionActivity, AFUserAgentFragment, AFRelatedMacro, AutoFocusAPI
AFSample.get("9040e9fda52931c9472c90ecad5b74295cdb9cf7b68e2b89219700f6a8bff5ac")`

The environment variable has to be set before the import.

**Comment by jeromeleonard** [comment]
thanks for your quick reply. ok :( It looks tricky to pass the configuration of the analyzer from Cortex to the program in this case.

**Comment by amr-cossi** [comment]
https://github.com/PaloAltoNetworks/autofocus-client-library/issues/19

**Pull request merged**